x-pack/plugins/observability/public/embeddable/slo/slo_overview.tsx

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useCallback, useEffect } from 'react';
+import { EuiIcon, EuiText, EuiLoadingSpinner } from '@elastic/eui';
+import { Chart, Metric, MetricTrendShape, Settings } from '@elastic/charts';
+import numeral from '@elastic/numeral';
+import { euiStyled } from '@kbn/kibana-react-plugin/common';
+import { ALL_VALUE } from '@kbn/slo-schema';
+import { i18n } from '@kbn/i18n';
+import { NOT_AVAILABLE_LABEL } from '../../../common/i18n';
+import { useKibana } from '../../utils/kibana_react';
+import { useFetchSloDetails } from '../../hooks/slo/use_fetch_slo_details';
+import { paths } from '../../../common/locators/paths';
+
+import { EmbeddableSloProps } from './types';
+
+export function SloOverview({ sloId, sloInstanceId, startTime, endTime }: EmbeddableSloProps) {
+  const {
+    uiSettings,
+    application: { navigateToUrl },
+    http: { basePath },
+  } = useKibana().services;
+  const { isLoading, slo, refetch, isRefetching } = useFetchSloDetails({
+    sloId,
+    instanceId: sloInstanceId,
+    shouldRefetch: true,
+  });
+
+  useEffect(() => {
+    refetch();
+  }, [startTime, endTime, refetch]);
+
+  const percentFormat = uiSettings.get('format:percent:defaultPattern');
+  const isSloNotFound = !isLoading && slo === undefined;
+
+  const getIcon = useCallback(
+    (type: string) =>
+      ({ width = 20, height = 20, color }: { width: number; height: number; color: string }) => {
+        return <EuiIcon type={type} width={width} height={height} fill={color} />;
+      },
+    []
+  );
+
+  if (isRefetching) {
+    return (
+      <LoadingPanel>
+        <LoadingContentPanel>
+          <EuiLoadingSpinner size="m" />
+          <EuiText>
+            <p>
+              {i18n.translate('xpack.observability.sloEmbeddable.widget.reloading', {
+                defaultMessage: 'Reloading',
+              })}
+            </p>
+          </EuiText>
+        </LoadingContentPanel>
+      </LoadingPanel>
+    );
+  }
+  if (isSloNotFound) {
+    return null;
+  }
+
+  const color =
+    slo?.summary.status === 'NO_DATA'
+      ? '#f8e9e9'
+      : slo?.summary.status !== 'HEALTHY'
+      ? '#f8e9e9'
+      : '#e6f9f7';
+
+  const extraContent = `Target <b>${numeral(slo?.objective.target).format(percentFormat)}</b>`;
+  // eslint-disable-next-line react/no-danger
+  const extra = <span dangerouslySetInnerHTML={{ __html: extraContent }} />;
```

Comment:
Is there a better way to do this? It looks dangerous.. The [Metric chart ](https://elastic.github.io/elastic-charts/?path=/story/metric-alpha--basic)is using it this way